app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,39 @@
+class ReviewsController < ApplicationController
+  before_action :set_user
+
+  def index
+    if params[:query].present?
+      author = User.where("name ILIKE ?", "%#{params[:query]}%").first
+      @reviews = Review.where(receiver: @user, author: author)
+    else
+      @reviews = Review.where(receiver: @user)
+    end
+  end
+
+  def new
+    @review = Review.new
```

Comment:
You should not just send to the view an empty new review, but a review with 
- review.barber_review = true/false based on which dashboard the http request comes from
- review.receiver = @user (defined in the :set_user method)